src/cli/cli_perf.cpp

```diff
@@ -0,0 +1,1790 @@
+    }
+    else
+    {
+        Server::sServer->OutputFormat("perf server ");
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagDestAddr))
+    {
+        Server::sServer->OutputFormat("destaddr %x:%x:%x:%x:%x:%x:%x:%x ", HostSwap16(destAddr->mFields.m16[0]),
+                                      HostSwap16(destAddr->mFields.m16[1]), HostSwap16(destAddr->mFields.m16[2]),
+                                      HostSwap16(destAddr->mFields.m16[3]), HostSwap16(destAddr->mFields.m16[4]),
+                                      HostSwap16(destAddr->mFields.m16[5]), HostSwap16(destAddr->mFields.m16[6]),
+                                      HostSwap16(destAddr->mFields.m16[7]));
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagBandwidth))
+    {
+        Server::sServer->OutputFormat("bandwidth %u ", aSetting.GetBandwidth());
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagLength))
+    {
+        Server::sServer->OutputFormat("length %u ", aSetting.GetLength());
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagInterval))
+    {
+        Server::sServer->OutputFormat("interval %u ", aSetting.GetInterval() / MS_PER_SEC);
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagFormatCvs))
+    {
+        Server::sServer->OutputFormat("format cvs ");
+    }
+    else if (aSetting.IsFlagSet(Setting::kFlagFormatQuiet))
+    {
+        Server::sServer->OutputFormat("format quiet ");
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagTime))
+    {
+        Server::sServer->OutputFormat("time %u ", aSetting.GetTime() / MS_PER_SEC);
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagNumber))
+    {
+        Server::sServer->OutputFormat("count %u ", aSetting.GetCount());
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagPriority))
+    {
+        Server::sServer->OutputFormat("priority %u ", aSetting.GetPriority());
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagSessionId))
+    {
+        Server::sServer->OutputFormat("id %u ", aSetting.GetSessionId());
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagFinDelay))
+    {
+        Server::sServer->OutputFormat("findelay %u ", aSetting.GetFinDelay());
+    }
+
+    if (aSetting.IsFlagSet(Setting::kFlagEcho))
+    {
+        Server::sServer->OutputFormat("echo %d ", aSetting.GetEchoFlag());
+    }
+
+    Server::sServer->OutputFormat("\r\n");
+}
+
+void Perf::SessionStop(uint8_t aRole)
+{
+    for (size_t i = 0; i < OT_ARRAY_LENGTH(mSessions); i++)
+    {
+        if (!mSessions[i].IsStateValid())
+        {
+            continue;
+        }
+
+        if (mSessions[i].GetRole() == aRole)
+        {
+            mSessions[i].Free();
+        }
+    }
+}
+
+otError Perf::ServerStart()
+{
+    otError  error = OT_ERROR_NONE;
+    Session *session;
+
+    VerifyOrExit(mServerSetting != NULL);
+    VerifyOrExit(!mServerRunning, error = OT_ERROR_BUSY);
+    VerifyOrExit((session = NewSession(*mServerSetting, Session::kRoleListener)) != NULL, error = OT_ERROR_NO_BUFS);
```

Comment:
I think we can use the heap for this purpose.

Reply:
Good idea. I will update the code to use heap.